packages/camera/camera_windows/windows/camera.h

```diff
@@ -0,0 +1,155 @@
+  virtual ~Camera() = default;
+
+  // Disallow copy and move.
+  Camera(const Camera &) = delete;
+  Camera &operator=(const Camera &) = delete;
+
+  virtual bool HasDeviceId(std::string &device_id) = 0;
+  virtual bool HasCameraId(int64_t camera_id) = 0;
+
+  virtual bool AddPendingResult(PendingResultType type,
+                                std::unique_ptr<MethodResult<>> result) = 0;
+  virtual bool HasPendingResultByType(PendingResultType type) = 0;
+
+  virtual camera_windows::CaptureController *GetCaptureController() = 0;
+
+  virtual void InitCamera(flutter::TextureRegistrar *texture_registrar,
+                          flutter::BinaryMessenger *messenger,
+                          bool enable_audio,
+                          ResolutionPreset resolution_preset) = 0;
+};
+
+class CameraImpl : public Camera {
+ public:
+  CameraImpl(const std::string &device_id);
+  virtual ~CameraImpl();
+
+  // Disallow copy and move.
+  CameraImpl(const CameraImpl &) = delete;
+  CameraImpl &operator=(const CameraImpl &) = delete;
+
+  // From CaptureControllerListener
+  void OnCreateCaptureEngineSucceeded(int64_t texture_id) override;
+  void OnCreateCaptureEngineFailed(const std::string &error) override;
+  void OnStartPreviewSucceeded(int32_t width, int32_t height) override;
+  void OnStartPreviewFailed(const std::string &error) override;
+  void OnPausePreviewSucceeded() override;
+  void OnPausePreviewFailed(const std::string &error) override;
+  void OnResumePreviewSucceeded() override;
+  void OnResumePreviewFailed(const std::string &error) override;
+  void OnStartRecordSucceeded() override;
+  void OnStartRecordFailed(const std::string &error) override;
+  void OnStopRecordSucceeded(const std::string &filepath) override;
+  void OnStopRecordFailed(const std::string &error) override;
+  void OnTakePictureSucceeded(const std::string &filepath) override;
+  void OnTakePictureFailed(const std::string &error) override;
+  void OnVideoRecordSucceeded(const std::string &filepath,
+                              int64_t video_duration) override;
+  void OnVideoRecordFailed(const std::string &error) override;
+  void OnCaptureError(const std::string &error) override;
+
+  // From Camera
+
+  bool HasDeviceId(std::string &device_id) override {
+    return device_id_ == device_id;
+  };
+
+  bool HasCameraId(int64_t camera_id) override {
+    return camera_id_ == camera_id;
+  };
+
+  bool AddPendingResult(PendingResultType type,
+                        std::unique_ptr<MethodResult<>> result) override;
+
+  bool HasPendingResultByType(PendingResultType type) override;
+
+  camera_windows::CaptureController *GetCaptureController() override {
+    return capture_controller_.get();
+  };
+
+  void InitCamera(flutter::TextureRegistrar *texture_registrar,
+                  flutter::BinaryMessenger *messenger, bool enable_audio,
+                  ResolutionPreset resolution_preset) override;
+
+  void InitCamera(
```

Comment:
All non-override methods need declaration comments, per style guide. (In at least some cases you already have declaration-sytle comments, they just need to be moved from the implementation file to here.)

Applies to all files in the PR.